indra/newview/lltooldraganddrop.cpp

```diff
@@ -320,6 +321,16 @@ void LLToolDragAndDrop::beginDrag(EDragAndDropType type,
 		LL_WARNS() << "Attempted to start drag without a cargo type" << LL_ENDL;
 		return;
 	}
+
+    if (type != DAD_CATEGORY)
+    {
+        LLViewerInventoryItem* item = dynamic_cast<LLViewerInventoryItem*>(gInventory.getObject(cargo_id));
```

Comment:
the cast is not needed, just call getItem directly

Reply:
The getItem returns LLInventoryItem*, may I be sure that this is exactly LLViewerInventoryItem* ?

Reply:
1. check what LLInventoryModel::getObject(uuid) does, it calls getCategory first, then getItem, but since you are making sure this isn't DAD_CATEGORY first half of getObejct is redundant, so it's better to just call getItem directly.
2. LLInventoryObject->LLInventoryItem->LLViewerInventoryItem in this case, you can use either of the three, since base class's LLInventoryObject::getUUID() will be used.